core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -649,24 +650,77 @@ String getFeatureVariableValueForType(@Nonnull String featureKey,
         String variableValue = variable.getDefaultValue();
         Map<String, ?> copiedAttributes = copyAttributes(attributes);
         FeatureDecision featureDecision = decisionService.getVariationForFeature(featureFlag, userId, copiedAttributes);
-        if (featureDecision.variation != null) {
+        Boolean featureEnabled = false;
+        if (featureDecision.variation != null && featureDecision.variation.getFeatureEnabled()) {
             FeatureVariableUsageInstance featureVariableUsageInstance =
                 featureDecision.variation.getVariableIdToFeatureVariableUsageInstanceMap().get(variable.getId());
             if (featureVariableUsageInstance != null) {
                 variableValue = featureVariableUsageInstance.getValue();
             } else {
                 variableValue = variable.getDefaultValue();
             }
+            featureEnabled = featureDecision.variation.getFeatureEnabled();
         } else {
             logger.info("User \"{}\" was not bucketed into any variation for feature flag \"{}\". " +
                     "The default value \"{}\" for \"{}\" is being returned.",
                 userId, featureKey, variableValue, variableKey
             );
         }
 
+        Object convertedValue = convertStringToType(variableValue, variableType);
```

Comment:
This conversion is now happening twice per api call. Once here and once in the originating method.